reactive/kubernetes_master.py

```diff
@@ -326,6 +326,28 @@ def install_snaps():
     remove_state('kubernetes-master.components.started')
 
 
+@when('etcd.available')
+@when('config.changed.enable-metrics')
+def enable_metric_changed():
+    """
+    Trigger an api server update.
+
+    :return: None
+    """
+    etcd = endpoint_from_flag('etcd.available')
+    connection = etcd.get_connection_string()
+    if not connection:
+        # etcd is not returning a connection string. This happens when
```

Comment:
I am concerned that we call configure_apiserver in a LOT of places in the code, but none of them have this check. Should the check be inside configure_apiserver instead so it catches it anywhere? I feel like we need to make a pass at unifying this and having an official way to handle reconfiguring. Maybe a flag that is cleared when it needs to happen. Seems that this code is copied all around and others may have attempted to start this with functions like reconfigure_cdk_addons.